torch/quantization/quantize.py

```diff
@@ -353,6 +353,13 @@ def swap_module(mod, mapping):
     if hasattr(mod, 'qconfig') and mod.qconfig is not None or type(mod) == DeQuantStub:
         if type(mod) in mapping:
             new_mod = mapping[type(mod)].from_float(mod)
+            # Preserve the module pre and post forward hooks except _observer_forward_hook
+            # Pre forward hooks will be called on quantized input
+            for pre_hook_fn in mod._forward_pre_hooks.values():
+                new_mod.register_forward_pre_hook(pre_hook_fn)
+            for hook_fn in mod._forward_hooks.values():
```

Comment:
sorry, this one as well, I guess
before convert:
  this will be executed before the observer _forward_hook
and after convert:
  this will be executed after the tensor has been quantized?

Reply:
feels like we are not doing things consistently in this case, maybe the observer_forward_hook should run before these hooks as well

Reply:
We have two options of calling observer_forward_hook:
1) When it's on Stub layer (that doesn't affect anything)
2) When it's on a non-stub layer (I mean Linear, Conv etc.) during static quantization

In the second case observer_forward_hook is registered for the examined layer first. And it will remain always first because of OrderedDict containing hooks

Reply:
No, this is wrong. 
During preparation step observer_forward_hook will be the last hook in a row, so it will take into account any change that other hooks have already introduced to output tensor.
When we'll convert this module, the final output will be quantized properly. But as for output from hooks that maybe passed to other layers I am not sure.
What do you think?

Reply:
Thanks for this PR!. One thing to note is that for functional_modules hooks will not work (for both floating point and quantized modules).
https://github.com/pytorch/pytorch/blob/master/torch/nn/quantized/modules/functional_modules.py

Reply:
Yes, you're right.
Functional modules keep their behaviour as they've had previously

Reply:
I think, the introduced code will preserve registered hooks on functional modules. But their forward is never called and they aren't meant to have hooks by construction